tools/eeprom-tool/cam96tof1_eeprom.cpp

```diff
@@ -0,0 +1,91 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "cam96tof1_eeprom.h"
+#include "eeprom_map.h"
+
+Camera96Tof1Eeprom::Camera96Tof1Eeprom(std::shared_ptr<aditof::EepromInterface> _eeprom):
+                    m_eeprom(_eeprom) {
+    if (!_eeprom){
+        LOG(ERROR) << "null pointer to eeprom.";
+    }
+}
+
+aditof::Status Camera96Tof1Eeprom::read(std::vector<uint8_t>& data){
+    float read_size = 100;
+    uint32_t eepromSize; 
+    aditof::Status status;
+    std::string eepromName;
+
+    m_eeprom->getName(eepromName);
+    eepromSize = EEPROMS.at(eepromName).size;
+
+    //mistery
+    m_eeprom->write(eepromSize - 5, (uint8_t *)&read_size, 4);
+
+    //read size
+    m_eeprom->read((uint32_t)0, (uint8_t *)&read_size, (size_t)4);
+    LOG(INFO) << "EEPROM calibration data size " << read_size << " bytes";
+    if (read_size > eepromSize) {
+        LOG(WARNING) << "Invalid calibration data size";
+        return aditof::Status::GENERIC_ERROR;
+    }
+
+    //read data
+    data.resize(read_size);
+    status = m_eeprom->read(4, data.data(), read_size);
+    if (status != aditof::Status::OK) {
+        data.resize(0);
+        LOG(WARNING) << "Failed to read from eeprom";
+        return status;
+    }
+
+    return aditof::Status::OK;
+}
+
+aditof::Status Camera96Tof1Eeprom::write(std::vector<uint8_t> data){
+    aditof::Status status;
+    float size = static_cast<float>(data.size());
+
+    status = m_eeprom->write((uint32_t)0, (uint8_t *)&size, (size_t)4);
+    if (status != aditof::Status::OK){
+        LOG(ERROR) << "failed to write to eeprom";
+        return status;
+    }
+
+    status = m_eeprom->write((uint32_t)4, (uint8_t *)data.data(), (size_t)size);
+    if (status != aditof::Status::OK){
+        LOG(ERROR) << "failed to write to eeprom";
+        return status;
+    }
+
+    return aditof::Status::OK;
+}
```

Comment:
Watch out for missing new lines at the end of file

Reply:
fixed